dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/tools/detectors/FailingTestDetector.java

```diff
@@ -74,8 +82,56 @@ private void output(final Set<String> tests) throws IOException {
             if (beforeSize == afterSize) {
                 break;
             }
+
+
         }
 
         return notPassingTests;
     }
-}
+
+    public static void outputFailedTests(MavenProject p, String testsPath, String outputPath){
```

Comment:
Minor: This method is a little long and has several components that could be factored out into small methods.